zipline/assets/_assets.pyx

```diff
@@ -75,58 +87,37 @@ cdef class Asset:
         """
         Cython rich comparison method.  This is used in place of various
         equality checkers in pure python.
-
-        <	0
-        <=	1
-        ==	2
-        !=	3
-        >	4
-        >=	5
         """
         cdef int x_as_int, y_as_int
 
-        if isinstance(x, Asset):
-            x_as_int = x.sid
-        elif isinstance(x, int):
-            # ints are Integral, but much faster to special case
-            x_as_int = x
-        elif isinstance(x, (long, np.int64, np.int32, Integral)):
-            x_as_int = int(x)
-        else:
+        try:
+            x_as_int = PyNumber_Index(x)
```

Comment:
That seems reasonable. The alternative would be to make this a bigger type like uint64 but idk if that is worth it and only moves the boundry up. The alternative is to catch this and return `NotImplemented` which will propagate as a type error. I think that error is less informative.